web/src/user_group_edit.js

```diff
@@ -278,6 +324,23 @@ export function handle_subgroup_edit_event(group_id) {
     if (is_editing_group(group_id)) {
         user_group_edit_members.update_member_list_widget(group);
     }
+
+    const tab_key = get_active_data().$tabs.first().attr("data-tab-key");
+    const subgroups_containing_current_user = direct_subgroup_ids.filter((group_id) =>
+        user_groups.is_user_in_group(group_id, people.my_current_user_id()),
+    );
+    // update display of group-rows on left panel.
+    // We need this update only if your-groups tab is active
+    // and current user is among the affect users as in that
+    // case the group widget list need to be updated and show
+    // or remove the group-row on the left panel accordingly.
```

Comment:
Given the code and comment duplication, maybe the function boundaries should be a bit different such that `update_your_groups_list` becomes `update_your_groups_list_if_needed` and the `tab_key` logic is part of it? Not sure, think about it and submit a PR if you think it'd help.